app/controllers/user.rb

```diff
@@ -0,0 +1,34 @@
+get '/login' do
+  erb :login
+end
+
+post '/login' do
+  @user = User.find_by(email: params[:user][:email])
+  if @user.authenticate(params[:user][:password])
```

Comment:
if @ user is nil (when email is not in database), it will throw an error for line 7. Something like "no method for NIL" class". So we have to first check that @user has something.

We can check for that by adding and "@user &&" before the "@user.authenticate"